pkg/config/file.go

```diff
@@ -33,42 +37,85 @@ import (
 var _ Loader = (*FileLoader)(nil)
 
 type FileLoader struct {
-	path     string
+	config   LoaderConfig
 	cRuntime chan<- runtime.Config
 	done     chan struct{}
+	fsys     fs.FS
 }
 
 func NewFileLoader(cfg *Config, cRuntime chan<- runtime.Config) *FileLoader {
 	return &FileLoader{
-		path:     cfg.Loader.File.Path,
+		config:   cfg.Loader,
 		cRuntime: cRuntime,
+		done:     make(chan struct{}, 1),
+		fsys:     os.DirFS(filepath.Dir(cfg.Loader.File.Path)),
 	}
 }
 
+// Run gets the runtime configuration from the local file.
+// The config will be loaded periodically defined by the loader interval configuration.
+// Returns an error if the loader is shutdown or the context is done.
 func (f *FileLoader) Run(ctx context.Context) error {
+	ctx, cancel := logger.NewContextWithLogger(ctx)
+	defer cancel()
 	log := logger.FromContext(ctx)
-	log.Info("Reading config from file", "file", f.path)
-	// TODO refactor this to use fs.FS
-	b, err := os.ReadFile(f.path)
+	tick := time.NewTicker(f.config.Interval)
```

Comment:
is there a reason you decided to go with polling over [fsnotify]?(https://github.com/fsnotify/fsnotify)

Reply:
No specific reason, I just learned of `fsnotify` right now 😄 

But imo we should do this as dedicated feature in the future since we already have the interval config field.

@puffitos What do you think about watching the file for changes instead of regularly checking the file with the set interval?

Reply:
Pff I don't know. Both approaches are valid.

The current approach uses the already used simple ticker logic we've been using in the whole repository. It works, it looks the same everywhere.

The fsnotify approach is simpler, but uses an experimental go library (x/sys). It's still being updated, but who knows what will happen to it :)

If you use fsnotify, you'll have to rewrite your tests, right? Do whatever works, but don't get perfect get in the way of done :)

Reply:
I'll stick to my implementation first. If we have the urge to change it, it can be done in another PR.